src/app/(subcategory)/subcategory-ai/hooks/useSetInitialCategory.ts

```diff
@@ -0,0 +1,22 @@
+import { type Dispatch, type SetStateAction, useEffect } from 'react';
+
+import { useSubCategoryContext } from '@app/contexts/SubCategoryContext';
+import type { CategoryType } from '@app/types/list.types';
+
+export function useSetInitialCategory(
+  category: CategoryType | null,
+  setCategory: Dispatch<SetStateAction<CategoryType | null>>,
+  categories: CategoryType[]
+) {
+  const { categoryId } = useSubCategoryContext();
+
+  useEffect(() => {
+    if (categoryId && !category) {
+      const selectedCategory = categories?.find((ct) => ct._id === categoryId);
+      if (selectedCategory) {
+        setCategory(selectedCategory);
+      }
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
```

Comment:
Is there another way to solve this rather than using eslint to fix this?

Reply:
I can set all parameters to array of dependencies. But I sujest I can decide by myself which dependencies should cause rerender. Thats why I turned off it for all project. But if it's not ok for you - I will chenge, no problems.

Reply:
I think it is better to build as React it expects, to avoid problems